python-avd/pyavd/api/ip_addressing/__init__.py

```diff
@@ -200,15 +201,19 @@ def p2p_uplinks_ip(self, uplink_switch_index: int) -> str:
 
         prefixlen = self.inputs.fabric_ip_addressing.p2p_uplinks.ipv4_prefix_length
         p2p_ipv4_pool, offset = self._get_p2p_ipv4_pool_and_offset(uplink_switch_index)
-
+        if p2p_ipv4_pool is None or offset is None:
+            msg = "Either p2p_ipv4_pool or offset is None"
```

Comment:
Maybe it is better to move this to the `_get_p2p_ipv4/6_pool_and_offset` and update the return type. Then raise separate errors for each key as needed. Also the AristaAvdMissingVariableError just takes a single variable name as argument.